src/main/java/TaskList.java

```diff
@@ -0,0 +1,248 @@
+            System.out.println("     Nice! I've marked this task as done:\n");
+            System.out.println("       "
+                    + "[" + tasks.get(ind).getStatusIcon() + "] "
+                    + tasks.get(ind).getDescription());
+        }
+
+    }
+
+    /**
+     * Marks specified task as done if taskNumber parameter is a string.
+     *
+     * @param taskNumber The 1th-index of task to be marked.
+     * @throws DukeException If taskNumber is out of bounds.
+     */
+    public void markTask(String taskNumber) throws DukeException {
+        int ind;
+        try {
+            ind = Integer.parseInt(taskNumber) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("Description must be a number.");
+        }
+
+
+        if(ind >= tasks.size() || ind < 0) {
+            throw new DukeException("Task " + (ind + 1) + " does not exist.");
+        }
+
+        tasks.get(ind).mark();
+
+        if(!isSilent) {
+            System.out.println("     Nice! I've marked this task as done:\n");
+            System.out.println("       "
+                    + "[" + tasks.get(ind).getStatusIcon() + "] "
+                    + tasks.get(ind).getDescription());
+        }
+
+    }
+
+    /**
+     * Marks specified task as not done.
+     *
+     * @param taskNumber The 1th-index of task to be marked.
+     * @throws DukeException If taskNumber is out of bounds.
+     */
+    public void unmarkTask(String taskNumber) throws DukeException {
+
+        int ind;
+        try {
+            ind = Integer.parseInt(taskNumber) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("Description must be a number.");
+        }
+
+        if(ind >= tasks.size() || ind < 0) {
+            throw new DukeException("Task " + (ind + 1) + " does not exist.");
+        }
+
+        tasks.get(ind).unmark();
+
+        if(!isSilent) {
+            System.out.println("     OK, I've marked this task as not done yet:\n");
+            System.out.println("       "
+                    + "[" + tasks.get(ind).getStatusIcon() + "] "
+                    + tasks.get(ind).getDescription());
+        }
+
+    }
+
+    /**
+     * Adds a todo task.
+     *
+     * @param todoTask The task to be added.
+     */
+    public void addTodo(String todoTask) throws DukeException {
+
+        tasks.add(new Todo(todoTask));
+
+        if(!isSilent) {
+            System.out.println("     Got it. I've added this task:");
+            System.out.println(tasks.get(tasks.size() -1));
+            System.out.println("     Now you have "
+                    + tasks.size() + (tasks.size() > 1 ? " tasks " : " task ")
+                    + "in the list.");
+        }
+
+    }
+
+    /**
+     * Adds a deadline task.
+     *
+     * @param todoTask The task to be added.
+     * @throws DukeException If incorrect arguments are provided.
+     */
+    public void addDeadline(String todoTask) throws DukeException {
```

Comment:
These methods can be extracted to belong to a Command class rather than TaskList class. 
TaskList is your in memory model, you shouldn't have to deal with split input and show messages to the user here.